### PR TITLE
[README] Added `Docker Usage`

### PR DESCRIPTION
**Pull request by shmuel44**
## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/XSUP-34370

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -215,14 +222,33 @@ Note that the following commands may work partially without an internet connecti
 
 ---
 
+## Docker Usage
+
+Docker is required to run certain commands effectively.
+
+The following commands require Docker:
```
Comment (ShirleyDenkberg) [inline README.md:229]:
```suggestion
The following command requires Docker:
```

File: README.md [hunk 2]
```diff
@@ -215,14 +222,33 @@ Note that the following commands may work partially without an internet connecti
...
 
 ## Custom Container Registry
 
 By default, the `demisto-sdk` will use `dockerhub` as the container registry to pull the integrations and scripts docker image.
 In order configure a custom container registry, the following environment variables must be set:
 
-* `DEMISTO_SDK_CONTAINER_REGISTRY`: the url of the container registry.
-* `DEMISTO_SDK_CR_USER`: the username to use in the container registry.
-* `DEMISTO_SDK_CR_PASSWORD`: the password to use in the container registry.
+- `DEMISTO_SDK_CONTAINER_REGISTRY`: the url of the container registry.
```
Comment (ShirleyDenkberg) [inline README.md:252]:
```suggestion
- `DEMISTO_SDK_CONTAINER_REGISTRY`: the URL of the container registry.
```

File: demisto_sdk/commands/generate_docs/README.md [hunk 3]
```diff
@@ -14,8 +16,8 @@ For integration - Top use-cases. Number the steps by '*' (i.e. '\* foo. * bar.')
 * **-c, --command**
 A comma-separated command names to generate doc for, will ignore the rest of the commands. e.g xdr-get-incidents,xdr-update-incident
 * **-e, --examples**
-- Path for a file containing examples. Each command should be in a separate line. or a comma separated list of commands
-- Comma separated list of examples, wrapped by quotes.
+Path for a file containing examples. Each command should be in a separate line. or a comma separated list of commands
```
Comment (ShirleyDenkberg) [inline demisto_sdk/commands/generate_docs/README.md:19]:
```suggestion
Path for a file containing examples. Each command should be in a separate line or a comma-separated list of commands.
```

File: demisto_sdk/commands/generate_docs/README.md [hunk 4]
```diff
@@ -14,8 +16,8 @@ For integration - Top use-cases. Number the steps by '*' (i.e. '\* foo. * bar.')
...
+Comma separated list of examples, wrapped by quotes.
```
Comment (ShirleyDenkberg) [inline demisto_sdk/commands/generate_docs/README.md:20]:
```suggestion
Comma-separated list of examples, wrapped by quotes.
```

**Comment by ShirleyDenkberg** [comment]
@dorschw Doc review completed.

**Review by dorschw — APPROVED** [review]

**Pull request merged**